### PR TITLE
Close #2884

### LINKED ISSUE #2884: Code Explorer should automatically add a `@Folder` annotation when modules are added.
I was watching @Xipooo's "Real World Demo of Using Interfaces" video and realized that we're missing a pretty big opportunity in the context menu.  We can tell pretty easily what folder in the Code Explorer is right-clicked to open the context menu:

screenshot from 2017-03-16 23 03 02

Why wouldn't we just add the  `@Folder("FolderThatWasRightClicked")` annotation to the new module? This would make organization a _lot_ easier as you're adding new code, and would be similar to how .NET and VS handle folders and namespaces.

### PR DESCRIPTION
**Pull request by Hosch250**
I don't like how these three commands each implement an identical `GetDeclaration` and `GetFolder`.

### CODE REVIEW

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddClassModuleCommand.cs [hunk 1]
```diff
@@ -57,7 +57,25 @@ private Declaration GetDeclaration(object parameter)
                 node = node.Parent;
             }
 
-            return node == null ? null : ((ICodeExplorerDeclarationViewModel)node).Declaration;
+            return ((ICodeExplorerDeclarationViewModel) node)?.Declaration;
+        }
+
+        private string GetFolder(object parameter)
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddClassModuleCommand.cs]: Nicely done :+1:

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddStdModuleCommand.cs [hunk 2]
```diff
@@ -33,20 +33,20 @@ protected override bool CanExecuteImpl(object parameter)
 
         protected override void ExecuteImpl(object parameter)
         {
+            var folderAnnotation = $"'@Folder(\"{GetFolder(parameter)}\")";
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddStdModuleCommand.cs]: Hmm.. Should we have a "default folder" setting? That way it wouldn't need to be duplicated in every module-creating command...

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs [hunk 3]
```diff
@@ -32,20 +32,20 @@ protected override bool CanExecuteImpl(object parameter)
 
         protected override void ExecuteImpl(object parameter)
         {
+            var folderAnnotation = $"'@Folder(\"{GetFolder(parameter)}\")";
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs]: We need to figure out a way to have this work when the module is added via standard VBE commands too... somehow.

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs [hunk 4]
```diff
@@ -57,7 +57,25 @@ private Declaration GetDeclaration(object parameter)
                 node = node.Parent;
             }
 
-            return node == null ? null : ((ICodeExplorerDeclarationViewModel)node).Declaration;
+            return ((ICodeExplorerDeclarationViewModel) node)?.Declaration;
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs]: Perhaps extract some *ModuleMaker* type that the commands would use for this? That way the relatively complex logic below would only be in one place?

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs (hunk 3 above)
Comment (Hosch250) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs]: Hooks should do it.

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs (hunk 4 above)
Comment (Hosch250) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddUserFormCommand.cs]: Makes sense.

**Pull request merged**